src/Selenium2Driver.php

```diff
@@ -1120,4 +1131,64 @@ private function trigger($xpath, $event, $options = '{}')
         $script = 'Syn.trigger("' . $event . '", ' . $options . ', {{ELEMENT}})';
         $this->withSyn()->executeJsOnXpath($xpath, $script);
     }
+
+    /**
+     * Uploads a file to the Selenium instance.
+     *
+     * Note that uploading files is not part of the official WebDriver
+     * specification, but it is supported by Selenium.
+     *
+     * @param string $path     The path to the file to upload.
+     *
+     * @return string          The remote path.
+     *
+     * @throws DriverException When PHP is compiled without zip support, or the file doesn't exist.
+     * @throws UnknownError    When an unknown error occurred during file upload.
+     * @throws \Exception      When a known error occurred during file upload.
+     *
+     * @see https://github.com/SeleniumHQ/selenium/blob/master/py/selenium/webdriver/remote/webelement.py#L533
+     */
+    public function uploadFile($path)
+    {
+        if (!is_file($path)) {
+          throw new DriverException('File does not exist locally and cannot be uploaded to the remote instance.');
+        }
+
+        if (!class_exists('ZipArchive')) {
+          throw new DriverException('Could not compress file, PHP is compiled without zip support.');
+        }
+
+        // Selenium only accepts uploads that are compressed as a Zip archive.
+        $temp_filename = tempnam('', 'WebDriverZip');
+
+        $archive = new \ZipArchive();
+        $archive->open($temp_filename, \ZipArchive::CREATE);
```

Comment:
you need to check the return value here, as ZipArchive uses the return value to handle errors instead of throwing exceptions